doi-transformer/src/main/java/no/unit/nva/doi/transformer/CrossRefConverter.java

```diff
@@ -165,46 +177,120 @@ private Range extractPages(CrossRefDocument document) {
         return StringUtils.parsePage(document.getPage());
     }
 
-    private BasicContext extractPublicationContext(CrossRefDocument document) throws InvalidIssnException {
+    private PublicationContext extractPublicationContext(CrossRefDocument document)
+            throws InvalidIssnException, InvalidIsbnException {
         CrossrefType crossrefType = CrossrefType.getByType(document.getType());
-        PublicationType publicationType = crossrefType.getPublicationType();
-
-        if (nonNull(publicationType) && publicationType.equals(PublicationType.JOURNAL_CONTENT)) {
-            // TODO actually call the Channel Register API and get the relevant details
-            return new Journal.Builder()
-                    .withLevel(null)
-                    .withTitle(extractJournalTitle(document))
-                    .withOnlineIssn(extractOnlineIssn(document))
-                    .withPrintIssn(extractPrintIssn(document))
-                    .withOpenAccess(false)
-                    .withPeerReviewed(false)
-                    .build();
-        } else {
-            throw new IllegalArgumentException(String.format(UNRECOGNIZED_TYPE_MESSAGE, document.getType()));
+        if (nonNull(crossrefType)) {
```

Comment:
Tidier with

        CrossrefType crossrefType = CrossrefType.getByType(document.getType());
        var publicationType = Optional.of(crossrefType.getPublicationType())
                .orElseThrow(() -> new IllegalArgumentException(MISSING_CROSSREF_TYPE_IN_DOCUMENT));

        if (publicationType.equals(PublicationType.JOURNAL_CONTENT)) {
            return createJournalContext(document);
        } else if (publicationType.equals(PublicationType.BOOK)) {
            return createBookContext(document);
        } else {
            throw new IllegalArgumentException(String.format(UNRECOGNIZED_TYPE_MESSAGE, document.getType()));
        }

Reply:
The second part can also be a function.  A function should do exactly one thing, i.e. it should contain one level of abstraction.

Reply:
Oh jess, cleaned up